service/build.gradle

```diff
@@ -33,54 +33,60 @@ apply from: 'analysis.gradle'
 
 dependencyManagement {
 	dependencies {
-		dependencySet(group:'io.github.hakky54', version: '7.3.0') {
-			entry 'sslcontext-kickstart-for-apache4'
+		dependencySet(group:'io.github.hakky54', version: '8.2.0') {
+			entry 'sslcontext-kickstart-for-apache5'
 			entry 'sslcontext-kickstart-for-pem'
 		}
 	}
 }
 
 dependencies {
-	annotationProcessor 'org.immutables:value:2.9.0'
-	implementation 'io.github.hakky54:sslcontext-kickstart-for-apache4'
+	annotationProcessor 'org.immutables:value:2.9.2'
+	implementation 'io.github.hakky54:sslcontext-kickstart-for-apache5'
 	implementation 'io.github.hakky54:sslcontext-kickstart-for-pem'
 
-	implementation('bio.terra:terra-common-lib:0.0.75-SNAPSHOT') {
+	implementation('bio.terra:terra-common-lib:0.1.9-SNAPSHOT') {
 		// remove transitive database deps, drshub has no db
 		exclude group: 'org.liquibase', module: 'liquibase-core'
 		exclude group: 'bio.terra', module: 'stairway-gcp'
 		exclude group: 'bio.terra', module: 'stairway'
 		exclude group: 'org.springframework.boot', module: 'spring-boot-starter-data-jdbc'
 		exclude group: 'com.google.oauth-client', module: 'google-oauth-client'
 	}
-	implementation 'org.broadinstitute.dsde.workbench:sam-client_2.13:0.1-1b67d8b'
-	implementation 'bio.terra:externalcreds-client-resttemplate:0.72.0-SNAPSHOT'
-	implementation 'io.sentry:sentry-logback:6.1.4'
-	implementation 'org.codehaus.janino:janino:3.1.7' // Provides if/else xml parsing for logback config
+	implementation 'org.broadinstitute.dsde.workbench:sam-client_2.13:0.1-6a32c36'
+	implementation 'bio.terra:externalcreds-client-resttemplate:0.111.0-SNAPSHOT'
```

Comment:
🚫 close but no cigar… 

This is the latest ECM client, but ECM does not yet publish a Jakarta client, which DrsHub needs in order to complete its migration.

I asked about this in Identiteam's [Slack](https://broadinstitute.slack.com/archives/C03GMG4DUSE/p1702418717623449) but may end up taking a crack at modifying ECM to publish such a client, following what @dvoet did in TDR: https://github.com/DataBiosphere/jade-data-repo/pull/1551

Currently, when I try to resolve a DRS URI associated with a passportable TDR snapshot, this triggers a call to the ECM client to fetch my passport, which fails exceptionally.  I think this is related to client incompatibility, but am not positive.
```
2023-12-12 17:21:53,428 INFO  [drshub-async-thread-1] b.terra.drshub.services.AuthService: Cache miss. Fetching passports from ECM
…
Caused by: java.lang.NoSuchMethodError: 'org.springframework.http.HttpStatus org.springframework.http.ResponseEntity.getStatusCode()'
        at bio.terra.externalcreds.client.ApiClient.invokeAPI(ApiClient.java:500)
        at bio.terra.externalcreds.api.OidcApi.getProviderPassportWithHttpInfo(OidcApi.java:314)
        at bio.terra.externalcreds.api.OidcApi.getProviderPassport(OidcApi.java:276)
        at bio.terra.drshub.services.AuthService.lambda$fetchPassports$12(AuthService.java:287)
…
```

Reply:
https://broadworkbench.atlassian.net/browse/DR-3397

Reply:
With the new Jakarta-based library, we can now resolve DRS URIs associated with passportable TDR snapshots:

Cache misses - call to ECM:
```
2023-12-13 12:05:30,707 INFO  [http-nio-8080-exec-2] b.t.d.c.DrsHubApiController: Received URL drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05 from agent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 on IP null
2023-12-13 12:05:30,716 INFO  [drshub-async-thread-1] b.t.d.services.DrsProviderService: Matched a hostname ID and stripped path: v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05
2023-12-13 12:05:30,721 INFO  [drshub-async-thread-1] b.t.d.services.DrsProviderService: built URI: drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05
2023-12-13 12:05:30,722 INFO  [drshub-async-thread-1] b.t.d.services.DrsResolutionService: Drs URI drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05 will use provider Terra Data Repo (TDR), requested fields bucket, contentType, fileName, gsUri, hashes, localizationPath, name, size, timeCreated, timeUpdated, googleServiceAccount
2023-12-13 12:05:30,723 INFO  [drshub-async-thread-1] b.t.drshub.services.DrsApiFactory: Cache miss. Creating RestTemplate for DRS Provider 'Terra Data Repo (TDR)'
2023-12-13 12:05:31,464 INFO  [drshub-async-thread-1] b.t.d.services.DrsResolutionService: Requesting DRS metadata for drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05 with auth required true from host jade.datarepo-dev.broadinstitute.org
2023-12-13 12:05:31,465 INFO  [drshub-async-thread-1] b.terra.drshub.services.AuthService: Cache miss. Fetching passports from ECM
2023-12-13 12:05:31,908 INFO  [drshub-async-thread-1] b.terra.drshub.services.AuthService: User does not have a passport.
2023-12-13 12:05:33,162 INFO  [drshub-async-thread-1] b.terra.drshub.logging.AuditLogger: DrsResolutionSucceeded
2023-12-13 12:05:33,188 INFO  [http-nio-8080-exec-2] b.t.drshub.logging.LoggerInterceptor: POST /api/v4/drs/resolve 200
```

Cache hits - no call to ECM:
```
2023-12-13 12:05:43,231 INFO  [http-nio-8080-exec-3] b.t.d.c.DrsHubApiController: Received URL drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05 from agent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 on IP null
2023-12-13 12:05:43,231 INFO  [drshub-async-thread-3] b.t.d.services.DrsProviderService: Matched a hostname ID and stripped path: v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05
2023-12-13 12:05:43,231 INFO  [drshub-async-thread-3] b.t.d.services.DrsProviderService: built URI: drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05
2023-12-13 12:05:43,231 INFO  [drshub-async-thread-3] b.t.d.services.DrsResolutionService: Drs URI drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05 will use provider Terra Data Repo (TDR), requested fields bucket, contentType, fileName, gsUri, hashes, localizationPath, name, size, timeCreated, timeUpdated, googleServiceAccount
2023-12-13 12:05:43,298 INFO  [drshub-async-thread-3] b.t.d.services.DrsResolutionService: Requesting DRS metadata for drs://jade.datarepo-dev.broadinstitute.org/v1_1a45eebc-ba3b-4336-8fb5-95c848f23893_57bbf476-b5ba-4ac0-8a79-fc1fc5e30b05 with auth required true from host jade.datarepo-dev.broadinstitute.org
2023-12-13 12:05:43,298 INFO  [drshub-async-thread-3] b.terra.drshub.services.AuthService: Cache hit. Not fetching passports from ECM
2023-12-13 12:05:43,882 INFO  [drshub-async-thread-3] b.terra.drshub.logging.AuditLogger: DrsResolutionSucceeded
2023-12-13 12:05:43,887 INFO  [http-nio-8080-exec-3] b.t.drshub.logging.LoggerInterceptor: POST /api/v4/drs/resolve 200
```

Sadly, it looks like we don't presently have RAS test accounts with passports so we are unable to check the case where a passport is actually used to resolve a DRS URI.  Above, we are falling back to resolving via bearer token.  But this is good enough for now.  Context: https://broadinstitute.slack.com/archives/C03GMG4DUSE/p1702487011872899